src/auto-complete.jsx

```diff
@@ -349,8 +355,9 @@ const AutoComplete = React.createClass({
     };
 
     const requestsList = [];
+    const maxSearchResults = this.props.maxSearchResults;
 
-    this.props.dataSource.map((item) => {
+    for (let item of this.props.dataSource) {
```

Comment:
Fine @oliviertassinari 
So I should use `Array.prototype.every` function or normal `for` to loop?